app/master/cluster_runner_config.py

```diff
@@ -160,7 +160,7 @@ def _shell_command_list_to_single_command(self, commands):
         # We should join the commands with double ampersands UNLESS the command already ends with a single ampersand.
         # A semicolon (or a double ampersand) is invalid syntax after a single ampersand.
         sanitized_commands = []
-        for command in commands:
+        for command in [c for c in commands if c is not None]:
```

Comment:
This is just kind of a style preference issue, but I wonder if this would be a bit more readable if we did the `None` check inside the existing `for` loop? I do like the list comprehension syntax but since we've already got a normal `for` loop here, it feels a bit weird to combine it with a comprehension.